typescript/ccip-server/src/services/LightClientService.ts

```diff
@@ -0,0 +1,90 @@
+// @ts-nocheck
+import axios from 'axios';
+import { ethers, utils } from 'ethers';
+
+import { TelepathyCcipReadIsmAbi } from '../abis/TelepathyCcipReadIsmAbi';
+
+import { Requestor } from './common/Requestor';
+
+enum ProofStatus {
+  running = 'running',
+  success = 'success',
+}
+
+// Service that interacts with the LightClient/ISM
+class LightClientService extends Requestor {
+  // Stores the current ProofId that is being generated. Clears once proof is ready.
+  pendingProofId: string;
+
+  constructor(
+    private readonly lightClientContract: ethers.Contract,
+    private readonly stepFunctionId: string,
+    private readonly chainId: string,
+    readonly platformUrl: string,
+    readonly platformApiKey: string,
+  ) {
+    super(axios, platformApiKey);
+  }
+
+  private getSyncCommitteePeriod(slot: BigInt): BigInt {
+    return slot / 8192n; // Slots Per Period
+  }
+
+  /**
+   * Gets syncCommitteePoseidons from ISM/LightClient
+   * @param slot
+   * @returns
+   */
+  async getSyncCommitteePoseidons(slot: BigInt): Promise<string> {
+    return await this.lightClientContract.syncCommitteePoseidons(
+      this.getSyncCommitteePeriod(slot),
+    );
+  }
+
+  /**
+   * Request the proof from Succinct.
+   * @param slot
+   * @param syncCommitteePoseidon
+   */
+  async requestProof(syncCommitteePoseidon: string, slot: BigInt) {
+    if (!this.pendingProofId) {
```

Comment:
does it make sense for the service to only handle one proof id at a time?

Reply:
might be nice if this was stateless

Reply:
As discussed offline, will add an object mapping to allow multiple proofs to be handled concurrently.